[13.0][ADD] sale_pwa_cache

Tardo opened the pull request:
Continue work from https://github.com/OCA/sale-workflow/pull/1365
Depends on:
 - https://github.com/OCA/product-attribute/pull/834
 - https://github.com/OCA/web/pull/1863

cc @tecnativa

pedrobaeza commented on sale_pwa_cache/views/sale_order_views.xml:
```diff
@@ -0,0 +1,158 @@
+                                context="{'company_id': parent.company_id}"
+                                class="mb-2"
+                                options="{'no_open': True, 'no_create': True, 'no_edit': True}"
+                                domain="[['category_id', '=', uom_category_id]]"
```
No need to force this domain, but use the standard one:

https://github.com/odoo/odoo/blob/c76454f325a52ba4bb49d9dae9f760046eb39ebd/addons/sale/models/sale.py#L1280

and add `product_uom_category_id` field.

pedrobaeza commented on sale_pwa_cache/views/sale_order_views.xml:
```diff
@@ -0,0 +1,158 @@
+                            }"
+                                context="{'company_id': parent.company_id}"
+                                class="mb-2"
+                                options="{'no_open': True, 'no_create': True, 'no_edit': True}"
```
Add `groups="uom.group_uom"`

Tardo committed:
[ADD] sale_pwa_cache

Tardo committed:
[IMP] sale_pwa_cache: Auto-confirm

Tardo committed:
[IMP] sale_pwa_cache: Translations

Tardo committed:
[IMP] sale_pwa_cache: Data

Tardo committed:
[IMP] sale_pwa_cache: Data

Tardo committed:
[IMP] sale_pwa_cache: Data and date_order

Tardo committed:
[IMP] sale_pwa_cache: Sync

pedrobaeza committed:
[IMP] sale_pwa_cache: Switch to noupdate=0

They are not going to be manually updated.

pedrobaeza committed:
[FIX] sale_pwa_cache: Missing trigger discriminant

pedrobaeza committed:
[IMP] sale_pwa_cache: Extra trigger for reducing scope of the update

After the change in the engine, we can add more than one trigger for
the same event and PWA cache, reducing the scope of the update that is
going to be performed.

Tardo committed:
[IMP] sale_pwa_cache: Add 'sale_ok' to group domains

pedrobaeza committed:
[FIX] sale_pwa_cache: Typos + proper PWA link

pedrobaeza committed:
[FIX+IMP] sale_pwa_cache: Remove useless PWA onchange definition + README

pedrobaeza committed:
[FIX] sale_pwa_cache: Missing .id

Tardo committed:
[IMP] sale_pwa_cache: Check disabled

Tardo committed:
[IMP] sale_pwa_cache: Add onchange for product_uom_category_id

Tardo committed:
[IMP] sale_pwa_cache: Avoid cache online

Tardo committed:
[IMP] sale_pwa_cache: Assign product type

sergio-teruel committed:
[IMP] sale_pwa_cache: Get history sale records with sudo to get sale lines from distinct salesperson
TT35287

Tardo committed:
[IMP] sale_pwa_cache: Force product_picker translations
